src/rgw/rgw_op.h

```diff
@@ -545,6 +546,10 @@ class RGWCreateBucket : public RGWOp {
 public:
   RGWCreateBucket() : has_cors(false) {}
 
+  void emplace_attr(std::string&& key, buffer::list&& bl) {
+    attrs.emplace(key, bl); /* key and bl are r-value refs */
```

Comment:
I don't find it convincing--but the whole point is to get the compiler to dtrt, so not relevant.